core/trie.go

```diff
@@ -144,6 +146,21 @@ func NewTrie(storage TrieStorage, height uint) *Trie {
 	}
 }
 
+// DoOnTempTrie creates an in-memory Trie of height `height` and runs `do` on that Trie
+func DoOnTempTrie(height uint, do func(*Trie) error) error {
+	db, err := db.NewInMemoryDb()
```

Comment:
```suggestion
	db, err := db.NewTestDb()
```

Reply:
same, DoOnTempTrie is not a test only function.